### PR TITLE
[cloud-provider-openstack] Fix ordering static nodes without security groups

### PR DESCRIPTION
**Pull request by vitaliy-sn**
Signed-off-by: Vitaliy Snurnitsin <vitaliy.snurnitsin@flant.com>

## Description

- Fixed ordering static nodes without security groups.
- Added the ability to set the disk type for CloudPermanent nodes.

## Why do we need it, and what problem does it solve?

Using nodeGroups it was not possible to create a node if the provider did not have support for security groups.

## What is the expected result?

Static nodes are created both with and without security groups.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-openstack
type: fix
summary: Fix ordering static nodes without security groups
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @vitaliy-sn. I've started the workflow for label 'e2e/run/openstack'...

File: ee/candi/cloud-providers/openstack/terraform-modules/static-node/main.tf [hunk 1]
```diff
@@ -4,6 +4,9 @@
 locals {
   actual_zones = lookup(var.providerClusterConfiguration, "zones", null) != null ? tolist(setintersection(data.openstack_compute_availability_zones_v2.zones.names, var.providerClusterConfiguration.zones)) : data.openstack_compute_availability_zones_v2.zones.names
   zones        = lookup(local.ng, "zones", null) != null ? tolist(setintersection(local.actual_zones, local.ng["zones"])) : local.actual_zones
+  volume_type_map      = var.providerClusterConfiguration.masterNodeGroup.volumeTypeMap
```
Comment (konstantin-axenov) [inline ee/candi/cloud-providers/openstack/terraform-modules/static-node/main.tf:7]: `masterNodeGroup`?

File: ee/candi/cloud-providers/openstack/openapi/cluster_configuration.yaml [hunk 2]
```diff
@@ -249,6 +249,21 @@ apiVersions:
               description: |
                 A limited set of zones in which nodes can be created.
               x-doc-required: false
+            volumeTypeMap:
+              description: |
+                A dictionary of disk types for root drive.
+
+                We recommend using the fastest disks provided by the provider in all cases.
```
Comment (konstantin-axenov) [inline ee/candi/cloud-providers/openstack/openapi/cluster_configuration.yaml:256]:
```suggestion
```

**Review by konstantin-axenov — CHANGES_REQUESTED** [review]

File: ee/candi/cloud-providers/openstack/terraform-modules/static-node/main.tf (hunk 1 above)
Comment (vitaliy-sn) [reply ee/candi/cloud-providers/openstack/terraform-modules/static-node/main.tf]: fixed

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @konstantin-axenov. I've started the workflow for label 'e2e/run/openstack'...

**Pull request merged**